pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -1945,8 +1945,13 @@ public boolean getBoolean(int columnIndex) throws SQLException {
       return false; // SQL NULL
     }
 
+    int col = columnIndex - 1;
+    if (Oid.BOOL == fields[col].getOID()) {
+      final byte[] value = this_row[col];
+      return (value.length == 1) && ('t' == (char) (value[0] & 0xFF));
```

Comment:
not quite the same :-) but close:
* `'t' == (char) (value[0] & 0xFF)`
* `(byte) 't' == value[0]`